src/antelope/stores/balances.ts

```diff
@@ -157,30 +157,63 @@ export const useBalancesStore = defineStore(store_name, {
                 // if the token is important, we always add it. Even with 0 balance.
-                return true;
+                result = true;
             } else {
-                return !balanceBn.isNegative() && !balanceBn.isZero();
+                result = !balanceBn.isNegative() && !balanceBn.isZero();
             }
+            this.trace('shouldAddTokenBalance', label, ethers.utils.formatUnits(balanceBn, token.decimals), token.symbol, '=>', result);
+            return result;
         },
         processBalanceForToken(label: string, token: EvmToken, balanceBn: BigNumber): void {
-            token.balanceBn = balanceBn;
-            token.balance = `${formatWei(balanceBn, token.decimals, 4)}`;
-            token.fullBalance = `${formatWei(balanceBn, token.decimals, token.decimals)}`;
-            if (token.price > 0) {
-                token.fiatBalance = `${parseFloat(token.balance) * token.price}`;
-            }
+            this.trace('processBalanceForToken', label, ethers.utils.formatUnits(balanceBn, token.decimals), token.symbol);
+            const balance = `${formatWei(balanceBn, token.decimals, 4)}`;
+            const fullBalance = `${formatWei(balanceBn, token.decimals, token.decimals)}`;
+            const fiatBalance = token.price > 0 ? `${parseFloat(balance) * token.price}` : '';
+
+            const tokenBalance = {
+                ...token,
+                balanceBn,
+                balance,
+                fullBalance,
+                fiatBalance,
+            } as EvmToken;
+
             if (this.shouldAddTokenBalance(label, balanceBn, token)) {
-                this.addNewBalance(label, token);
+                this.addNewBalance(label, tokenBalance);
+            } else {
+                this.removeBalance(label, tokenBalance);
+            }
+        },
+        async subscribeForTransactionReceipt(response: TransactionResponse): Promise<TransactionResponse> {
+            this.trace('subscribeForTransactionReceipt', response.hash);
+            const provider = toRaw(useEVMStore().rpcProvider);
+            if (provider) {
+                // instead of await, we use then() to return the response immediately
+                // and perform the balance update in the background
+                provider.waitForTransaction(response.hash).then((receipt: ethers.providers.TransactionReceipt) => {
+                    this.trace('subscribeForTransactionReceipt', response.hash, 'receipt:', receipt.status, receipt);
+                    if (receipt.status === 1) {
+                        const account = useAccountStore().loggedAccount;
+                        if (account?.account) {
+                            this.updateBalancesForAccount('logged', account);
+                        }
+                        // TODO: should we notify the user that the transaction succeeded?
+                    } else {
+                        // TODO: should we do something if the transaction failed?
```

Comment:
can you just make an issue to track these todos? for user feedback for transaction status